### ISSUE TITLE
Old file share gets user whole owncloud content

### ISSUE THREAD

**Issue by dicenl** [owncloud/core#16919]
### Steps to reproduce
I wasnt able to reproduce. Al I can say is that I have upgraded from owncloud 8.0 to 8.1 to 8.2. to 8.3

### Expected behaviour
Old file shares shouldnt be active after the share is deleted and the folder/files deleted from owncloud

### Actual behaviour
Last week i had a big security issue at one of my clients. A user shared a single folder with 1 txt file in owncloud with an external user, this was months ago. I believe it was a single file in a folder. A couple of days after the file was shared the share was unshared and the folder and file deleted. Now last week the external user needed the file again, looked at his old emails and re-used the old link in the hope it would stil work. After clicking the download link and using the share password a download.zip started to download and he went on with his work to check the zip later. The zip file did not contain the txt file he needed but the complete owncloud content (about 400 MB). 

https://forum.owncloud.org/viewtopic.php?f=31&t=28769&p=89428#p89428

### Server configuration
**Operating system**: CentOS 6

**Web server: Apache 2

**Database: MySQL

**PHP version: 5.4

**ownCloud version: 8.3

**Updated from an older ownCloud or fresh install: 
updated from 8.0 to 8.1. to 8.2 to 8.3

**List of activated apps: 
Activity
Deleted files
First Run Wizard
LDAP user and group backend
PDF viewer
Share Files
Updater
ownCloud dependencies info

**The content of config/config.php:**

```php
<?php
$CONFIG = array (
  'instanceid' => 'occa9a748465',
  'passwordsalt' => '',
  'trusted_domains' => 
  array (
    0 => 'owncloud.xxx.xx',
  ),
  'datadirectory' => '/var/www/html/data',
  'dbtype' => 'mysql',
  'version' => '8.0.3.4',
  'dbname' => 'xxx_ocdb',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxx_ocdb',
  'dbpassword' => '',
  'installed' => true,
  'forcessl' => true,
  'ldapIgnoreNamingRules' => false,
  'theme' => '',
  'maintenance' => false,
  'loglevel' => '2',
  'secret' => '',
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No external storage. Just local storage on the server

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP – ActiveDirectory

#### LDAP configuration (delete this part if not used)

```
run: sqlite3 data/owncloud.db
then execute:  select * from oc_appconfig where appid='user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.

SQLite version 3.6.20 
Enter ".help" for instructions 
Enter SQL statements terminated with a ";" 
sqlite> select * from oc_appconfig where appid='user_ldap'; 
Error: no such table: oc_appconfig 
sqlite> 
```

### Client configuration
**Browser:**
Internet explorer

**Operating system:**
Windows 7

#### Web server error log
```
[Sun Jun 07 04:15:19 2015] [notice] Digest: generating secret for digest authentication ...
[Sun Jun 07 04:15:19 2015] [notice] Digest: done
[Sun Jun 07 04:15:20 2015] [notice] Apache/2.2.15 (Unix) DAV/2 PHP/5.4.41 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips configured -- resuming normal operations
[Sun Jun 07 14:01:45 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Mon Jun 08 06:32:25 2015] [error] [client x.x.x.x] client denied by server configuration: /usr/share/phpMyAdmin/scripts
[Mon Jun 08 09:00:43 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/dnscfg.cgi
[Mon Jun 08 15:35:18 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Mon Jun 08 15:36:56 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Mon Jun 08 20:34:57 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/images
[Tue Jun 09 13:38:07 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Tue Jun 09 13:38:10 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Tue Jun 09 13:41:01 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Tue Jun 09 20:01:27 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/payment
[Wed Jun 10 08:18:01 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/admin
[Wed Jun 10 09:39:19 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/muieblackcat
[Wed Jun 10 09:39:20 2015] [error] [client x.x.x.x] client denied by server configuration: /usr/share/phpMyAdmin/scripts
[Wed Jun 10 09:39:20 2015] [error] [client x.x.x.x] client denied by server configuration: /usr/share/phpMyAdmin/scripts
[Wed Jun 10 09:39:20 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/pma
[Wed Jun 10 09:39:20 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/myadmin
[Wed Jun 10 09:39:21 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/MyAdmin
[Wed Jun 10 15:07:22 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Wed Jun 10 17:58:21 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/html
[Thu Jun 11 02:07:35 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/payment
[Thu Jun 11 12:27:57 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Thu Jun 11 12:27:58 2015] [error] [client x.x.x.x] script '/var/www/html/testproxy.php' not found or unable to stat
[Thu Jun 11 18:13:36 2015] [error] [client x.x.x.x] client denied by server configuration: /usr/share/phpMyAdmin
[Thu Jun 11 18:13:36 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/favicon.ico
[Thu Jun 11 18:13:36 2015] [error] [client x.x.x.x] File does not exist: /var/www/html/favicon.ico
[Thu Jun 11 18:14:14 2015] [notice] caught SIGTERM, shutting down
[Thu Jun 11 18:14:15 2015] [notice] suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Thu Jun 11 18:14:15 2015] [notice] Digest: generating secret for digest authentication ...
[Thu Jun 11 18:14:15 2015] [notice] Digest: done
[Thu Jun 11 18:14:16 2015] [notice] Apache/2.2.15 (Unix) DAV/2 PHP/5.4.41 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips configured -- resuming normal operations
[Thu Jun 11 18:14:48 2015] [notice] caught SIGTERM, shutting down
[Thu Jun 11 18:14:48 2015] [notice] suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Thu Jun 11 18:14:48 2015] [notice] Digest: generating secret for digest authentication ...
[Thu Jun 11 18:14:48 2015] [notice] Digest: done
[Thu Jun 11 18:14:48 2015] [notice] Apache/2.2.15 (Unix) DAV/2 PHP/5.4.41 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips configured -- resuming normal operations
[Thu Jun 11 18:19:14 2015] [notice] caught SIGTERM, shutting down
[Thu Jun 11 18:19:14 2015] [notice] suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Thu Jun 11 18:19:14 2015] [notice] Digest: generating secret for digest authentication ...
[Thu Jun 11 18:19:14 2015] [notice] Digest: done
[Thu Jun 11 18:19:14 2015] [notice] Apache/2.2.15 (Unix) DAV/2 PHP/5.4.41 mod_ssl/2.2.15 OpenSSL/1.0.1e-fips configured -- resuming normal operations
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"edb5d253b7e584c05771a5b7eeacb538","remoteAddr":"x.x.x.216","app":"webdav","message":"Exception: {\"Message\":\"expected filesize 369043 got 98304\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(84): OC_Connector_Sabre_File->put(Resource id #356)\\n#1 \\\/var\\\/www\\\/html\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(1647): OC_Connector_Sabre_Directory->createFile('ISOWD1 19291 SR...', Resource id #356)\\n#2 \\\/var\\\/www\\\/html\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(900): Sabre\\\\DAV\\\\Server->createFile('documents\\\/028\\\/S...', Resource id #356, NULL)\\n#3 [internal function]: Sabre\\\\DAV\\\\Server->httpPut('documents\\\/028\\\/S...')\\n#4 \\\/var\\\/www\\\/html\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(474): call_user_func(Array, 'documents\\\/028\\\/S...')\\n#5 \\\/var\\\/www\\\/html\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Server.php(214): Sabre\\\\DAV\\\\Server->invokeMethod('PUT', 'documents\\\/028\\\/S...')\\n#6 \\\/var\\\/www\\\/html\\\/apps\\\/files\\\/appinfo\\\/remote.php(61): Sabre\\\\DAV\\\\Server->exec()\\n#7 \\\/var\\\/www\\\/html\\\/remote.php(54): require_once('\\\/var\\\/www\\\/html\\\/a...')\\n#8 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php\",\"Line\":130}","level":4,"time":"2015-06-07T22:22:06+00:00"}
{"reqId":"3800cff4b0618a7bb97c1b020e231df6","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T06:20:51+00:00"}
{"reqId":"3800cff4b0618a7bb97c1b020e231df6","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T06:20:51+00:00"}
{"reqId":"3800cff4b0618a7bb97c1b020e231df6","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T06:20:51+00:00"}
{"reqId":"3800cff4b0618a7bb97c1b020e231df6","remoteAddr":"x.x.x.147","app":"core","message":"Login failed: 'pgoosk' (Remote IP: 'x.x.x.147', X-Forwarded-For: '')","level":2,"time":"2015-06-08T06:20:51+00:00"}
{"reqId":"9edfc2d6bd9c2fd2c26cea1d217fcdda","remoteAddr":"x.x.x.73","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T06:40:51+00:00"}
{"reqId":"9edfc2d6bd9c2fd2c26cea1d217fcdda","remoteAddr":"x.x.x.73","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T06:40:51+00:00"}
{"reqId":"9edfc2d6bd9c2fd2c26cea1d217fcdda","remoteAddr":"x.x.x.73","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T06:40:51+00:00"}
{"reqId":"9edfc2d6bd9c2fd2c26cea1d217fcdda","remoteAddr":"x.x.x.73","app":"core","message":"Login failed: 'tlinde' (Remote IP: 'x.x.x.73', X-Forwarded-For: '')","level":2,"time":"2015-06-08T06:40:51+00:00"}
{"reqId":"a60d01742d86da92ef96444edcf95291","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T07:25:06+00:00"}
{"reqId":"a60d01742d86da92ef96444edcf95291","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T07:25:07+00:00"}
{"reqId":"a60d01742d86da92ef96444edcf95291","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T07:25:07+00:00"}
{"reqId":"a60d01742d86da92ef96444edcf95291","remoteAddr":"x.x.x.147","app":"core","message":"Login failed: 'bavene' (Remote IP: 'x.x.x.147', X-Forwarded-For: '')","level":2,"time":"2015-06-08T07:25:07+00:00"}
{"reqId":"4b861653b1773a84fe3bdf000d58b12e","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T07:47:09+00:00"}
{"reqId":"4b861653b1773a84fe3bdf000d58b12e","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T07:47:09+00:00"}
{"reqId":"4b861653b1773a84fe3bdf000d58b12e","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T07:47:09+00:00"}
{"reqId":"4b861653b1773a84fe3bdf000d58b12e","remoteAddr":"x.x.x.147","app":"core","message":"Login failed: 'mgugte' (Remote IP: 'x.x.x.147', X-Forwarded-For: '')","level":2,"time":"2015-06-08T07:47:09+00:00"}
{"reqId":"a0d382587e41d415da24ad6c83a472cc","remoteAddr":"x.x.x.198","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T19:07:00+00:00"}
{"reqId":"a0d382587e41d415da24ad6c83a472cc","remoteAddr":"x.x.x.198","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T19:07:00+00:00"}
{"reqId":"a0d382587e41d415da24ad6c83a472cc","remoteAddr":"x.x.x.198","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-08T19:07:00+00:00"}
{"reqId":"a0d382587e41d415da24ad6c83a472cc","remoteAddr":"x.x.x.198","app":"core","message":"Login failed: 'bavene' (Remote IP: 'x.x.x.198', X-Forwarded-For: '')","level":2,"time":"2015-06-08T19:07:00+00:00"}
{"reqId":"562fc30bb907cf07a6ccd625a058dd51","remoteAddr":"x.x.x.154","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T06:51:37+00:00"}
{"reqId":"562fc30bb907cf07a6ccd625a058dd51","remoteAddr":"x.x.x.154","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T06:51:37+00:00"}
{"reqId":"562fc30bb907cf07a6ccd625a058dd51","remoteAddr":"x.x.x.154","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T06:51:37+00:00"}
{"reqId":"562fc30bb907cf07a6ccd625a058dd51","remoteAddr":"x.x.x.154","app":"core","message":"Login failed: 'tlinde' (Remote IP: 'x.x.x.154', X-Forwarded-For: '')","level":2,"time":"2015-06-09T06:51:37+00:00"}
{"reqId":"c9487b934e82a8ad04275b67eecd0b69","remoteAddr":"x.x.x.73","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T06:57:36+00:00"}
{"reqId":"c9487b934e82a8ad04275b67eecd0b69","remoteAddr":"x.x.x.73","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T06:57:36+00:00"}
{"reqId":"c9487b934e82a8ad04275b67eecd0b69","remoteAddr":"x.x.x.73","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T06:57:36+00:00"}
{"reqId":"c9487b934e82a8ad04275b67eecd0b69","remoteAddr":"x.x.x.73","app":"core","message":"Login failed: 'bavene' (Remote IP: 'x.x.x.73', X-Forwarded-For: '')","level":2,"time":"2015-06-09T06:57:36+00:00"}
{"reqId":"4f68e5a468f2dd855274a4257ed5589a","remoteAddr":"x.x.x.76","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T21:11:23+00:00"}
{"reqId":"4f68e5a468f2dd855274a4257ed5589a","remoteAddr":"x.x.x.76","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T21:11:23+00:00"}
{"reqId":"4f68e5a468f2dd855274a4257ed5589a","remoteAddr":"x.x.x.76","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-09T21:11:23+00:00"}
{"reqId":"4f68e5a468f2dd855274a4257ed5589a","remoteAddr":"x.x.x.76","app":"core","message":"Login failed: 'bavene' (Remote IP: 'x.x.x.76', X-Forwarded-For: '')","level":2,"time":"2015-06-09T21:11:23+00:00"}
{"reqId":"27aa136fffb410e586d8cded1f8c2120","remoteAddr":"x.x.x.198","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T07:09:40+00:00"}
{"reqId":"27aa136fffb410e586d8cded1f8c2120","remoteAddr":"x.x.x.198","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T07:09:40+00:00"}
{"reqId":"27aa136fffb410e586d8cded1f8c2120","remoteAddr":"x.x.x.198","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T07:09:40+00:00"}
{"reqId":"27aa136fffb410e586d8cded1f8c2120","remoteAddr":"x.x.x.198","app":"core","message":"Login failed: 'mgugte' (Remote IP: 'x.x.x.198', X-Forwarded-For: '')","level":2,"time":"2015-06-10T07:09:40+00:00"}
{"reqId":"9e4334f9b6b209e53afae1e9723a96fa","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T14:33:35+00:00"}
{"reqId":"9e4334f9b6b209e53afae1e9723a96fa","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T14:33:35+00:00"}
{"reqId":"9e4334f9b6b209e53afae1e9723a96fa","remoteAddr":"x.x.x.147","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T14:33:35+00:00"}
{"reqId":"9e4334f9b6b209e53afae1e9723a96fa","remoteAddr":"x.x.x.147","app":"core","message":"Login failed: 'Lmout' (Remote IP: 'x.x.x.147', X-Forwarded-For: '')","level":2,"time":"2015-06-10T14:33:35+00:00"}
{"reqId":"26d92b55480f0e0a36b96739182f36f1","remoteAddr":"x.x.x.171","app":"user_ldap","message":"Configuration Error (prefix s01): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T20:20:59+00:00"}
{"reqId":"26d92b55480f0e0a36b96739182f36f1","remoteAddr":"x.x.x.171","app":"user_ldap","message":"Configuration Error (prefix s02): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T20:20:59+00:00"}
{"reqId":"26d92b55480f0e0a36b96739182f36f1","remoteAddr":"x.x.x.171","app":"user_ldap","message":"Configuration Error (prefix s03): either no password is given for theuser agent or a password is given, but not anLDAP agent.","level":2,"time":"2015-06-10T20:20:59+00:00"}
{"reqId":"26d92b55480f0e0a36b96739182f36f1","remoteAddr":"x.x.x.171","app":"core","message":"Login failed: 'Lmout' (Remote IP: 'x.x.x.171', X-Forwarded-For: '')","level":2,"time":"2015-06-10T20:20:59+00:00"}
```
#### Browser log

Dont have it. It was of an external user / supplier

```

**Comment by LukasReschke** [comment]
Thanks for reporting issues back to us. First of all there never was a "8.3" or "8.1" release yet which would be a new major release, I assume you refer to "8.0.3" and "8.0.1".

Please try to reproduce if this problem persists in 8.0.3 with the same sharing link. There has been some changes made to the code which should prevent this behaviour.

**Comment by LukasReschke** [comment]
Ref https://github.com/owncloud/core/pull/15145

**Comment by LukasReschke** [comment]
Steps to reproduce:
- Login as user "admin"
- Create a subfolder "1"
  - Create a subfolder "2" in "1"
    - Create a subfolder "3" in "2"
      - Share folder "3" publicly and upload a single folder to it
        - Copy the link
- In a private tab access the public link, all files there as expected
- Delete folder "1" and also delete it from the trashbin
- Access the public link, download the ZIP will download all information

=< 8.0.2: :boom: 
>= 8.0.3: Downloading is not possible anymore, a proper error is thrown.

@dicenl Please verify if this is the case that you experienced.

**Comment by dicenl** [comment]
Sorry, I indeed mean 8.0.0 to 8.0.1 to 8.0.2 to 8.0.3 Thinking about it, I think I started upgrading from 7.0.4 as I had to disable all apps before upgrading

After reproducing the steps as discriped below. I stil could download the complete owncloud content.

1. create folder named 1
2. create folder named 2 in 1
3. create folder named 3 in 2
4. upload file to folder 3
5. share folder 3 publicly
6. check public link (see file as expected)
7. delete folder 1
8. check public link again
9. see empty thumbnail with download button, but not working
10. delete files from trashbin
11. check public link. Now there is no thumbnail but there is a download button wich downloads complete owncloud content.

However, if I test it on an other owncloud server (my own) it does not work like discribed above. On my own server I get File not found The specified document has not been found on the server. This server was installed from version 8.0.0 and uses an external /data disk

![screenshot from 2015-06-14 13 55 27](https://cloud.githubusercontent.com/assets/6197489/8148431/867f7a70-129d-11e5-982a-a7955557101c.png)
![screenshot from 2015-06-14 13 55 05](https://cloud.githubusercontent.com/assets/6197489/8148428/867cd2de-129d-11e5-9c84-eddc7c687e66.png)
![screenshot from 2015-06-14 13 54 14](https://cloud.githubusercontent.com/assets/6197489/8148430/867f590a-129d-11e5-8938-e5874529a353.png)
![screenshot from 2015-06-14 13 53 52](https://cloud.githubusercontent.com/assets/6197489/8148429/867e6338-129d-11e5-9199-ba95feb15657.png)

**Comment by dicenl** [comment]
screenshots should be read in reverse order

**Comment by LukasReschke** [comment]
On which exact versions did you test step 11? Did this include 8.0.3?

**Comment by dicenl** [comment]
yes its on 8.0.3
![version](https://cloud.githubusercontent.com/assets/6197489/8148557/bb6138fa-12a2-11e5-88bf-f82bf0cd1aee.png)

**Comment by LukasReschke** [comment]
I'm not able to reproduce the steps on 8.0.3 instances, would it be possible for you to create a not-administrative ownCloud account on the server for me? You can find my current mail address in my profile.

I might come back to you then with SQL queries which we would need the results of to analyze this problem further.

**Comment by dicenl** [comment]
ok. check your email

**Comment by LukasReschke** [comment]
Can you please compare whether the file `apps/files_sharing/lib/controllers/sharecontroller.php` on your instance equals exactly to https://github.com/owncloud/core/blob/v8.0.3/apps/files_sharing/lib/controllers/sharecontroller.php? Alternatively you might just replace the file with the file from our Git repository.

If this does not work then please insert the following at line 160 (https://github.com/owncloud/core/blob/v8.0.3/apps/files_sharing/lib/controllers/sharecontroller.php#L160):

```php
\OC::$server->getLogger()->critical('$originalSharePath: '. print_r($originalSharePath, true), ['app' => $this->appName]);
\OC::$server->getLogger()->critical('$path: '. print_r($path, true), ['app' => $this->appName]);
\OC::$server->getLogger()->critical('isReadable: '. print_r(Filesystem::isReadable($originalSharePath . $path), true), ['app' => $this->appName]);
```

Once you have done these changes please try to access `/index.php/s/5q8332L0m9OSuea` on your instance and reported back the errors in the owncloud log.

**Comment by dicenl** [comment]
i checked the sharecontroller.php file with the one on github. they were not exactly the same. i have sent the files to your email account. Now i have replaced the php file all seems to be working fine again. do you have any idea what went wrong? 

there were no new messages in the owncloud log file.

**Comment by LukasReschke** [comment]
Interesting. Can you let me know how you installed the update and from where you got it?

**Comment by dicenl** [comment]
i got the update from owncloud.org and i used the owncloud-8.0.3.tar.bz2

for the 7.0.4 to 8.0.0 update i first had to disable all apps that were installed so i only got files left. then in the /tmp folder of the server i untar'd 

/tmp/# tar -xf owncloud-8.0.3.tar.bz2
/tmp/# cd owncloud
/tmp/# \cp * -R /var/www/html
/tmp/# chown apache:apache /var/www/html -R

this is how i updated every version 7.0.4 to 8.0.3 and al in between.

**Comment by LukasReschke** [comment]
Ok. This is just a crazy assumption but could it be that you only replaced the files in the root folder of ownCloud and not the subdirectories when updating?

This would mean that it shows 8.0.3 because this is defined in the root folder in `version.php`. Can you check if other files in subfolders do not equal to the ones from https://github.com/owncloud/core/tree/v8.0.3? This would strengthen this theory.

**Comment by dicenl** [comment]
The following files are checked and compared with the ownCloud instance running and the ownCloud files from the owncloud-8.0.3.tar.bz2 from the owncloud.org website and are al identical

/apps/files/lib/app.php
/core/application.php
/core/setup/controller.php
/lib/public/files/storage.php

I used the history in bash to check what commands were used and found out that the last update by .tar.bz2 was version 7.0.4 I suspect the rest of the updates were done by the owncloud updater. In the bash history I found the oldest verion installed was 6.0.4, so sorry for that mis information. I got a lot of different owncloud servers at different customers so I apparently dont remember al updates. 

bottom line, it looks like most files were done correctly. do you have any specific files i could compare?

**Comment by LukasReschke** [comment]
> bottom line, it looks like most files were done correctly. do you have any specific files i could compare?

Unfortunately the only reliable way is to compare all files or copy all over them. - That should be the safest way. 

> I suspect the rest of the updates were done by the owncloud updater. In the bash history I found the oldest verion installed was 6.0.4, so sorry for that mis information

Good to hear. THX for the feedback. – I will close this issue as the original issue seems to be addressed.

@VicDeo @karlitschek Do we have something in place to prevent this half-baken upgrades in the future? This is really bad as it gives the user a wrong impression and may impose security risks.

**Comment by dicenl** [comment]
Tnx for your support Lukas!! It would indeed be nice if there were some more checks than only the version.php file. maybe a file create time stamp check or file size check. Tomorrow i will manually overwrite al files (8.0.3) check if al is working fine and then a manual update tot 8.0.4

**Comment by karlitschek** [comment]
@LukasReschke no idea how to do this. I think this can't happen with rpms. Not sure how a check can be done when used manual tar upgrade.
@VicDeo Any idea what can be done in the updater?

**Comment by LukasReschke** [comment]
The update from 8.0.2 to 8.0.3 has happened using the updater app.

**Comment by dicenl** [comment]
yes the updates from 8.0.0 and up to 8.0.3 were with the updater app. all updates before that were by tar.bz2

**Comment by RealRancor** [comment]
@karlitschek @LukasReschke 

SMF for example has a maintenance task "Check all files against current versions" (http://wiki.simplemachines.org/smf/SMF2.0:Forum_Maintenance)

Each file has an header with the version like: https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Admin.php#L13

When running this task each file is checked against the current version of SMF (like the version.php in oC) and a warning is thrown if a file wasn't updated.
